deploy/libproject/deploy.rb

```diff
@@ -0,0 +1,207 @@
+require "open3"
+require "set"
+require_relative "../../aou-utils/serviceaccounts"
+require_relative "../../aou-utils/utils/common"
+require_relative "../../aou-utils/workbench"
+require_relative "../../api/libproject/devstart"
+require_relative "../../api/libproject/gcloudcontext"
+require_relative "../../api/libproject/wboptionsparser"
+
+
+DOCKER_KEY_FILE_PATH = "/creds/sa-key.json"
+
+def get_source_project(p)
+  # Returns the project preceding "p" in the release promotion process.
```

Comment:
It might be better to always use staging rather than using the preceding version... I can imagine a case where we need a quick fix to prod and don't wait for a push to stable to complete beforehand.

Reply:
Oops, forgot to reply to this one, but good suggestion - done.

It does leave a slightly awkward scenario where you're using this script to deploy to staging; handled that explicitly.